DigitalLearningSolutions.Data.Tests/Services/SessionServiceTests.cs

```diff
@@ -3,80 +3,106 @@
     using System;
     using System.Linq;
     using System.Transactions;
-    using DigitalLearningSolutions.Data.Models;
     using DigitalLearningSolutions.Data.Services;
     using DigitalLearningSolutions.Data.Tests.Helpers;
     using FluentAssertions;
+    using Microsoft.AspNetCore.Http;
     using NUnit.Framework;
 
     internal class SessionServiceTests
     {
         private SessionService sessionService;
         private SessionTestHelper sessionTestHelper;
+        private ISession httpContextSession;
 
         [SetUp]
         public void Setup()
         {
             var connection = ServiceTestHelper.GetDatabaseConnection();
             sessionService = new SessionService(connection);
             sessionTestHelper = new SessionTestHelper(connection);
+            httpContextSession = new MockHttpContextSession();
         }
 
         [Test]
-        public void StartOrRestartSession_Should_Start_Users_First_Session()
+        public void StartOrUpdateSession_Should_Start_Users_First_Session()
         {
             using (new TransactionScope())
             {
+                // Given
+                httpContextSession.Clear();
+
                 // When
                 const int candidateId = 29;
                 const int customisationId = 100;
-                var sessionId = sessionService.StartOrRestartSession(candidateId, customisationId);
+                sessionService.StartOrUpdateSession(candidateId, customisationId, httpContextSession);
 
                 // Then
-                var newSession = sessionTestHelper.GetSession(sessionId);
+                httpContextSession.Keys.Should().BeEquivalentTo($"SessionID-{customisationId}");

```

Comment:
It's preferable to have as few assertions as possible per test. This makes the tests easier to understand and means if they're broken then it should be easier to identify why. So I'd suggest testing that the http context session has been set correctly separately from testing that the session in the db has been updated correctly 🙂 If it would make it easier you could even split the session service up so you'd have:
* SessionDataService (or similar) which is equivalent to the session service you had before introducing the http context session
* SessionService which calls the SessionDataService methods and also handles the http context session

What do you think? I think keeping it all in one service is also fine if you prefer, as long as you split the tests up so each test is only testing one thing 🙂

Reply:
Yeah I agree that's a much better way of doing it! I've refactored to `SessionDataService`/`SessionService` approach in e4f7ed2 🙂